### PR TITLE
destroy action

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description
A small lesson on the `destroy` action.  

## Feedback Requested
Please utilize the built-in GitHub Reviewers functionality to assign necessary reviewers. Use this section to further specify portions of the PR that you would like additional or focused feedback on.

### CODE REVIEW

File: 08-rails/destroy-action.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+1. routes, and using route params
+1. Adding a link to the destroy action's route
+1. updating the controller
+1. writing two tests: one for nominal case, and one for edge case.
+
+### Using Route Parameters
+
+Open up the `BooksController` - we will need to add a new method `destroy` like this:
+
```
Comment (droberts-ada) [inline 08-rails/destroy-action.md]: Are you missing the route?

File: 08-rails/destroy-action.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+## Learning Goals
+
+- Understand the responsibility of the Rails convention `destroy` action
+- Review the nominal and edge cases when testing the `destroy` action
+- Use a route parameter in a controller action
+- Learn to make a `delete` request
+
+## The `destroy` Action
+
+The `destroy` action is the last controller action we will define. Its responsibility is, given some information (through route params), find a specific item, and then delete it from the database. **It is the _Delete_ in CRUD.** The `destroy` action is a _Rails convention_, so we will continue to use the name `destroy`.
```
Comment (droberts-ada) [inline 08-rails/destroy-action.md]: Maybe "the last of the RESTful controller actions we will define"?

File: 08-rails/destroy-action.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Testing
+
+Earlier when we looked at [testing database changes in controllers](`./testing-db-in-controllers.ms`) we examined a test for the `destroy` action.  
+
```
Comment (droberts-ada) [inline 08-rails/destroy-action.md]: This link is incorrect

Comment (droberts-ada) [reply 08-rails/destroy-action.md]: Also, does this mean they'll have seen tests for `destroy` before we get to this lesson? That doesn't feel right to me - I suspect it will be an awkward flow to the week and be confusing to the students.

Looking at [that section of the testing db in controllers lesson](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/08-rails/testing-db-in-controllers.md#content), it might be the fix is to change that to focus exclusively on `create`.

File: 08-rails/code_samples/destroy_controller_tests.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+describe "destroy" do
+  describe "destroy" do
+    it "can destroy a model" do
+      # Arrange
+      poodr = Book.new title: "Practical Object Oriented Programming in Ruby", author: "Sandi Metz"
```
Comment (kaidamasaki) [inline 08-rails/code_samples/destroy_controller_tests.rb]: This is a little hard to read without parens.

```suggestion
      poodr = Book.new(title: "Practical Object Oriented Programming in Ruby", author: "Sandi Metz")
```

File: 08-rails/code_samples/destroy_controller_tests.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+      poodr.save
+      id = poodr.id
+
+      # Act
+      expect {
+        delete book_path(id)
+
+        # Assert
+      }.must_change "Book.count", -1
```
Comment (kaidamasaki) [inline 08-rails/code_samples/destroy_controller_tests.rb]: Should these be `do` `end` blocks or is `expect` weird?

Comment (CheezItMan) [reply 08-rails/code_samples/destroy_controller_tests.rb]: Just continuing the pattern we established in regular rails testing.

File: 08-rails/destroy-action.md (hunk 3 above)
Comment (CheezItMan) [reply 08-rails/destroy-action.md]: Updated
